scripts/migration/migrate_share_registration_data.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import sys
+import django
+from django.db import transaction
+django.setup()
+
+from osf.models import Registration
+from scripts import utils as script_utils
+from website import settings
+from website.app import init_app
+from website.project.tasks import on_registration_updated
+
+logger = logging.getLogger(__name__)
+
+def migrate(dry_run):
+    assert settings.SHARE_URL, 'SHARE_URL must be set to migrate.'
+    assert settings.SHARE_API_TOKEN, 'SHARE_API_TOKEN must be set to migrate.'
+    registrations = Registration.objects.all()
+    registrations_count = registrations.count()
+    count = 0
+
+    logger.info('Preparing to migrate {} registrations.'.format(registrations_count))
+    for registration in registrations.iterator():
+        if not registration.is_public or registration.is_deleted:
+            continue
+        count += 1
```

Comment:
This should be above the `continue` otherwise you'll end up with some add looking counts. 
Or, better idea, just filter them out in the query.